dot/core/service.go

```diff
@@ -496,6 +497,31 @@ func (s *Service) InsertKey(kp crypto.Keypair) {
 	s.keys.Insert(kp)
 }
 
+// HasKey returns true if given hex encoded public key string is found in keystore, false otherwise, error if there
+//  are issues decoding string
+func (s *Service) HasKey(pubKeyStr string, keyType string) (bool, error) {
```

Comment:
this would probably be better as a function in the keystore package that can be wrapped by `core`

Reply:
Good idea, updated.